### ISSUE TITLE
Errors relating to LTI after merging 2.14

### ISSUE THREAD

**Issue by Alex-Jordan** [openwebwork/webwork2#878]
I merged the 2.14 release candidate branch into my production branch, and trying to visit any WeBWorK page gives this warning and error:

```
Warning messages

    database reference is not defined at /opt/webwork/webwork2/lib/WeBWorK/Authen/LTIAdvanced/SubmitGrade.pm line 50.

Error messages

    Could not create WeBWorK::Authen::LTIAdvanced::SubmitGrade object, missing items from request at /opt/webwork/webwork2/lib/WeBWorK/ContentGenerator.pm line 174. 
```

Is there anyone with a more firm understanding of the LTI modules (@dsteinmo ?) that could shed light on this? At this point I am unsure if it is an issue that has anything to do with the particulars of my setup (some customization done to my webwork2 and pg repos, and we are using Desire2Learn).

**Comment by dlglin** [comment]
The error is happening in the code that handles the scheduled LTI pass back of grades to the LMS.  You should be able to temporarily get past this by turning off grade pass back: comment out $LTIGradeMode in authentication_LTI.conf.

I’m working on getting 2.14 on my production server in the next day.  I’ll see if I can reproduce the issue.

On Aug 22, 2018, at 3:17 PM, Alex Jordan <notifications@github.com<mailto:notifications@github.com>> wrote:


I merged the 2.14 release candidate branch into my production branch, and trying to visit any WeBWorK page gives this warning and error:

Warning messages

    database reference is not defined at /opt/webwork/webwork2/lib/WeBWorK/Authen/LTIAdvanced/SubmitGrade.pm line 50.

Error messages

    Could not create WeBWorK::Authen::LTIAdvanced::SubmitGrade object, missing items from request at /opt/webwork/webwork2/lib/WeBWorK/ContentGenerator.pm line 174.


Is there anyone with a more firm understanding of the LTI modules (@dsteinmo<https://github.com/dsteinmo> ?) that could shed light on this? At this point I am unsure if it is an issue that has anything to do with the particulars of my setup (some customization done to my webwork2 and pg repos, and we are using Desire2Learn).

**Comment by mgage** [comment]
The database object $db is not being passed in to the SubmitGrade object constructor.  
Perhaps a configuration problem?  I'd follow Danny's suggestion, perhaps even comment out 
the include() statement in localOverides.conf
/Volumes/WW_test/opt/webwork/webwork2_2018/conf/localOverrides.conf:441: #include("conf/authen_LTI.conf");

and make sure things  work without the LTI module included.

**Comment by Alex-Jordan** [comment]
Seems to have been fixed by #880.
